### PR TITLE
Agent Modes — Supervisor Topology (Deep Research)

### PR DESCRIPTION
**Pull request by shenjianan97**
## Summary

Ships the **Supervisor Topology (Deep Research)** track for Agent Modes (S1–S11) plus post-S11 hardening. A supervisor agent decomposes a research brief, fans out isolated sub-agents over a shared checkpointer, and a Writer synthesizes a cited report. §A12 deferred-decisions ledger is fully dispositioned (D1–D3 Closed with metrics).

This branch is the track merge (26 commits ahead of `main`). The three most recent commits are the post-S11 follow-ups verified in this session:

### Post-S11 commits
1. **`subagent_completed` terminal success marker** — closes a visibility gap where a sub-agent that succeeded with **zero findings** emitted nothing after `subagent_started`, stranding the Console badge on "running". Adds the success counterpart to `subagent_failed` (bracket `subagent_started → subagent_completed | subagent_failed`). Worker emit + migration `0026` (lock-safe `NOT VALID`→`VALIDATE`→rename) + API projection (`marker.subagent.completed`, first-wins dedup) + Console (`done` status, "Completed" leaf).
2. **Writer citation hardening** — inline per-claim `[finding_id]` citation is now MANDATORY and the Writer must not author its own Sources section (the runtime appends `## Sources` from inline ids). Regression for the 2026-06-14 run where citations were batched at the end.
3. **Preserve immutable `agent_config` fields on Console update** — a Console edit of a supervisor agent omits topology/preset/supervisor; the gate now triggers only on an explicit topology change and inherits the immutable fields on absence, so an edit no longer rejects or wipes the agent to react.

## Testing

- **Worker:** writer-prompt unit (3), DB-backed `subagent_events` + `supervisor_fanout` (35, incl. migration `0026` admission), e2e `subagent_fanout_durability` + `supervisor_fanout_budget` (4).
- **Java:** `ActivityProjectionServiceTest` + `AgentServiceTest` — BUILD SUCCESSFUL.
- **Console:** full vitest suite 199 passed (incl. `ActivityPane` 26).
- **Browser gate (Scenario 21):** verified against a live Deep Research task —
  - §5: sub-agent tree, round header "Round 1 · 4 sub-agents · decision: stop", `done` chips, finding leaves with `finding_id` + clickable `source_url`, "Completed" leaves.
  - §5d: zero-finding subtask renders **"done"** + "Completed" leaf, not stuck "running".
  - AgentService fix: editing the Deep Research agent and saving preserves `topology=supervisor` / `preset=research` / supervisor config.

## Migration

`0026_subagent_completed_event.sql` admits the `subagent_completed` event_type (25→26). Must land in prod before any worker build that emits it (deploy-order §A6).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: services/api-service/src/main/java/com/persistentagent/api/service/AgentService.java [hunk 1]
```diff
@@ -237,6 +323,125 @@ private String serializeConfig(AgentConfigRequest config) {
...
+                : readPersistedString(persistedConfigRaw, "preset");
+        SupervisorConfigRequest supervisor = request.supervisor() != null
+                ? request.supervisor()
+                : readPersistedSupervisor(persistedConfigRaw);
+        return new AgentConfigRequest(
+                request.systemPrompt(),
+                request.provider(),
+                request.model(),
+                request.temperature(),
+                request.allowedTools(),
```
[the remark this answers is not in the document]
Comment (shenjianan97) [reply services/api-service/src/main/java/com/persistentagent/api/service/AgentService.java]: Fixed in 1586eae. Confirmed the gap: `canonicalizeConfig` re-admits preset-injected names only when present in `allowed_tools`, and the Console echoes only the user-facing allowlist on update — so once the preset was preserved but its hidden tools weren't re-sent, a coding/investigation agent lost `dispatch_subagent` on every edit. `inheritImmutableFields` now re-derives the inherited preset's injected tools (`PresetDefaults.injectedToolsForPreset`) and merges them back before canonicalisation. Regression test added (coding preset omits hidden tools → dispatch_subagent survives). research is idempotent (web tools are user-facing); no-op for the no-preset path.

File: services/worker-service/executor/graph.py [hunk 2]
```diff
@@ -3661,6 +4216,92 @@ async def run_astream():
...
+        model_name = agent_config.get("model", "claude-3-5-sonnet-latest")
+        temperature = agent_config.get("temperature", 0.7)
+        model = await providers.create_llm(
+            self.pool, provider, model_name, temperature
+        )
+
+        # The sub-agents' tool objects — the agent's own built-in tools (the
+        # ``research`` preset seeds the web allowlist). The headless filter inside
+        # ``run_subagent`` drops any interrupt-bearing tool before binding.
+        allowed_tools = agent_config.get("allowed_tools", [])
```
[the remark this answers is not in the document]
Comment (shenjianan97) [reply services/worker-service/executor/graph.py]: Fixed in 146a34c. The fan-out built `sub_tools` from the full `allowed_tools` and never read `supervisor.source_allowlist`. Added `_filter_subagent_source_tools`: a non-empty allowlist drops web source tools (`web_search`/`read_url`) not named in it, applied before `_get_tools`. Empty/absent = all sources; base platform tools the sub-agent needs are never filtered. v1 governs the built-in web sources only — sub-agents receive no tool-server tools yet, so a tool-server name in the allowlist has no built-in tool to gate. Tests cover the helper edge cases + end-to-end filtering through `_inject` (uncheck read_url → sub-agents get only web_search).

File: services/worker-service/executor/graph.py [hunk 3]
```diff
@@ -3661,6 +4216,92 @@ async def run_astream():
...
+        sub_tools = self._get_tools(
+            allowed_tools,
+            cancel_event=asyncio.Event(),
```
[the remark this answers is not in the document]
Comment (shenjianan97) [reply services/worker-service/executor/graph.py]: Fixed in 146a34c. The fan-out passed a throwaway `asyncio.Event()` into `_get_tools` that no task code sets, unlike the ReAct path. Threaded the task's real `cancel_event` through `execute_task` → `_inject_supervisor_configurable` → `_get_tools`, so an in-flight sub-agent web/tool call now aborts on cancellation/lease-loss instead of running (and spending) until the next super-step boundary. Regression test asserts `_get_tools` receives the real event (sentinel identity).
